ONSdigital/dp-design-system
# Feature/port header footer

### What

- Adding styling and JS required for the header and footer
- Changed local port number to avoid conflict with `dp-compose/cantabular-import` minio service

### How to review

To run the changes locally you'll need to:
- Pull and run this branch
- Use a service that has been ported to use the `dp-renderer`; I recommend using a cantabular dataset landing page - there are [full instructions](https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import) on how to setup your mac to view these pages
   - If using the above ⬆️, you'll need to use the `replace` statement in `dp-frontend-dataset-controller` := follow these [instructions](https://github.com/ONSdigital/dp-renderer#referencing-a-local-instance-of-dp-renderer-in-a-docker-container)
- Update [dp-renderer/assets/templates/main.tmpl](https://github.com/ONSdigital/dp-renderer/blob/main/assets/templates/main.tmpl) to point to your local instance of the `dp-design-system`; I would recommend removing references to `Sixteens` in the template to have confidence that the changes are not dependent upon Sixteens

`main.tmpl` would look like something like this: 
```html
    {{ if .FeatureFlags.ONSDesignSystemVersion }}
      <link rel="stylesheet" href="http://localhost:9002/dist/assets/css/main.css">
    {{ end }}
...
    {{ if .FeatureFlags.ONSDesignSystemVersion }}
        </div>
      </div>
      <script src="http://localhost:9002/dist/assets/js/main.js"></script>
    {{end}}

```

- Once you have a page running, visually it should appear as the main site and the JS interactions should work as they currently do
- N.B. `dp-design-system` uses vanilla JS (no jQuery) unlike Sixteens; there appears to be a bug with `es5` bundle so testing in IE11 will result in errors. This bug is being actively worked on.

### Who can review

Frontend dev
